modules/@apostrophecms/schema/ui/apos/components/AposInputArray.vue

```diff
@@ -71,7 +195,67 @@ export default {
       if (result) {
         this.next = result;
       }
+    },
+    getNext() {
+      // Next should consistently be an array.
+      return (this.value && Array.isArray(this.value.data))
+        ? this.value.data : (this.field.def || []);
+    },
+    disableAdd() {
+      return this.field.max && (this.items.length >= this.field.max);
+    },
+    remove(_id) {
+      this.items = this.items.filter(color => color._id !== _id);
+    },
+    add() {
+      this.items.push({
+        _id: cuid(),
+        schemaInput: {
+          data: this.newInstance()
+        }
+      });
+    },
+    newInstance() {
+      const instance = {};
+      for (const field of this.field.schema) {
+        if (field.def !== undefined) {
+          instance[field.name] = klona(field.def);
+        }
+      }
+      return instance;
     }
   }
 };
+
+function expandItems(items) {
+  return items.map(item => ({
+    _id: item._id || cuid(),
+    schemaInput: {
+      data: item
+    }
+  }));
+}
 </script>
+<style lang="scss" scoped>
+  .apos-is-dragging {
+    opacity: 0.5;
+    background: var(--a-base-4);
+  }
+  .apos-input-array-inline-item {
+    display: flex;
+    margin-bottom: 20px;
+    border-left: 1px solid var(--a-base-9);
+  }
+  .apos-input-array-inline-item-controls {
+    // align-self: stretch seems like the answer, but looks bad
+    // once the field adds additional height to itself for an error.
+    // 15px - 7.5px (padding of icons) = 7.5px
+    padding-top: 7.5px;
+  }
+  .apos-input-array-inline-schema-wrapper {
+    flex-grow: 1;
+  }
+  .apos-drag-handle {
+    padding: 7.5px;
```

Comment:
are we ok with pixels as floats?

Reply:
I'm borrowing from existing practice elsewhere, 7.5px appears quite a bit in this situation.